reports/GitalyFail/GitalyFail.tla

```diff
@@ -0,0 +1,137 @@
+------------------------------ MODULE GitalyFail ------------------------------
+(* 
+    MIT License Copyright (c) 2020 CQFN
+    https://github.com/cqfn/degitx/blob/master/LICENSE
+    
+    DeGit Tech Report. 
+    Appendix 1. Gitaly fails explore.
+           
+ *)
+===============================================================================
+CONSTANT RM
+
+VARIABLES rmState, tmState, tmPrepared, 
+    msgs, rmGenerations, tmGeneration, 
+
+Messages == [type: {"Prepared"}, rm: RM] \union [type:{"Commit", "Abort"}]
+
+
+TPTypeOK == /\ rmState \in [RM -> {"working", "prepared", "pre-committed", "committed", "aborted"}]
+            /\ tmState \in {"init", "done"}
+            /\ tmPrepared \subseteq RM
+            /\ msgs \subseteq Messages
+            
+TPInit ==   /\ rmState = [r \in RM |-> "working"]
+            /\ tmState = "init"
+            /\ tmPrepared = {}
+            /\ msgs = {}            
+            /\ rmGenerations = [r \in RM |-> 0]
+            /\ tmGeneration = 0
+            /\ rmPrepared = {},
+            /\ rmCommitted = {}
+            /\ rmAborted = {}
+
+\* Don't call - endless loop
+TMSend(r) == /\ \E r \in RM
+             /\ rmState[r] = "working"
+             /\ RMRcvPrep(r)
+             /\ UNCHANGED << rmState, tmState, msgs >> 
+
+(*
+    Expected stages 
+    
+    TMInit + RMInit (all)
+    TMsendPrepare
+    RMRcvPrep -
+        RMIgnorePrep
+        RMDoPrepare
+            RMConfirmPrepare - TMRcvConfirmation
+    TMCalculateQuorum - add rms to voted
+    TMSendCommit - to voted rms
+    TMSendAbort - to voted rms
+    RMRvcAbort
+    RMRcvCommit -
+        RMIgnoreCommit
+        RMDoCommit
+            RMConfirmCommit - add rms to confirmed commits
+    TMIncreaseGen for confirmed
+    
+*)
+
+DoNothing == /\ UNCHANGED << tmState, rmState >>
+
+RMRcvPrep(r) == /\ \E r \in RM 
+                /\ rmState[r] == "worknig"
+                /\ (\/ DoNothing
+                    \/ RMDoPrepare(r) )
```

Comment:
It's not correct: RM can either prepare or abort here. On abort it changes self state from workfing to aborted and ignoring following messages.